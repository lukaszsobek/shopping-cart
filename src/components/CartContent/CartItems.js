import React, { Component } from "react";
import CartItem from "./CartItem";

import { deleteItem } from "../../actions";

import { connect } from "react-redux";

class CartItems extends Component {

    render() {
        const { cartContent, deleteItem } = this.props;

        if(!cartContent.length) {
            return (
                <div className="no-items">
                    Add an item to the cart
                </div>
            )
        }

        return cartContent.map((item, key) =>{
            return (
                <CartItem
                    key={ key }
                    id={ key }
                    number={ key + 1}
                    name={ item.name }
                    comments={ item.comments }
                    net_price={ item.net_price }
                    tax={ item.tax }
                    deleteItem={ deleteItem }
                />
            );
        });

    }
};

function mapStateToProps(state) {
    return {
        cartContent: state.cartContent
    }
}

function mapDispatchToProps(dispatch) {
    return {
        deleteItem: item => dispatch(deleteItem(item))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(CartItems);