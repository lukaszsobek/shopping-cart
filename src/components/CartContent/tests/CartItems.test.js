import CartItems from "../CartItems";

describe("Renders cart",() => {

    const item = {
        name: "test item",
        comments: "none",
        net_price: 12.12,
        tax: 7
    }

    it("without items", () => {
        const props = {
            cartContent: [],
            deleteItem: () => null,
            openEditItemModal: () => null
        }
        expect(CartItems(props)).toMatchSnapshot();
    });

    it("with 1 item", () => {
        const props = {
            cartContent: [ item ],
            deleteItem: () => null,
            openEditItemModal: () => null
        }
        expect(CartItems(props)).toMatchSnapshot();
    });

    it("with 2 items", () => {
        const props = {
            cartContent: [ item, item ],
            deleteItem: () => null,
            openEditItemModal: () => null
        }
        expect(CartItems(props)).toMatchSnapshot();
    });
});

