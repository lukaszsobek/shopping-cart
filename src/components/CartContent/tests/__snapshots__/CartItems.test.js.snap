// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders cart with 1 item 1`] = `
Array [
  <CartItem
    comments="none"
    deleteItem={[Function]}
    id={0}
    name="test item"
    net_price={12.12}
    number={1}
    openEditItemModal={[Function]}
    tax={7}
/>,
]
`;

exports[`Renders cart with 2 items 1`] = `
Array [
  <CartItem
    comments="none"
    deleteItem={[Function]}
    id={0}
    name="test item"
    net_price={12.12}
    number={1}
    openEditItemModal={[Function]}
    tax={7}
/>,
  <CartItem
    comments="none"
    deleteItem={[Function]}
    id={1}
    name="test item"
    net_price={12.12}
    number={2}
    openEditItemModal={[Function]}
    tax={7}
/>,
]
`;

exports[`Renders cart without items 1`] = `
<div
  className="no-items"
>
  Add an item to the cart
</div>
`;
